MsCorePkg/Library/SecureBootKeyStoreLib/SecureBootKeyStoreLib.c

```diff
@@ -0,0 +1,100 @@
+/** @file SecureBootKeyStoreLib.c
+
+
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <UefiSecureBoot.h>
+#include <Guid/ImageAuthentication.h>
+#include <Library/SecureBootVariableLib.h>
+
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PcdLib.h>
+
+#define PLATFORM_SECURE_BOOT_KEY_COUNT  2
+
+SECURE_BOOT_PAYLOAD_INFO  *gSecureBootPayload     = NULL;
+UINT8                     gSecureBootPayloadCount = 0;
+
+UINT8                     mSecureBootPayloadCount                            = PLATFORM_SECURE_BOOT_KEY_COUNT;
+SECURE_BOOT_PAYLOAD_INFO  mSecureBootPayload[PLATFORM_SECURE_BOOT_KEY_COUNT] = {
+  {
+    .SecureBootKeyName = L"Microsoft Only",
+    .KekPtr            = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultKek),
+    .KekSize           = (CONST UINT32)FixedPcdGetSize (PcdDefaultKek),
+    .DbPtr             = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultDb),
+    .DbSize            = (CONST UINT32)FixedPcdGetSize (PcdDefaultDb),
+    .DbxPtr            = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultDbx),
+    .DbxSize           = (CONST UINT32)FixedPcdGetSize (PcdDefaultDbx),
+    .PkPtr             = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultPk),
+    .PkSize            = (CONST UINT32)FixedPcdGetSize (PcdDefaultPk),
+    .DbtPtr            = NULL,
+    .DbtSize           = 0,
+  },
+  {
+    .SecureBootKeyName = L"Microsoft Plus 3rd Party",
+    .KekPtr            = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultKek),
+    .KekSize           = (CONST UINT32)FixedPcdGetSize (PcdDefaultKek),
+    .DbPtr             = (CONST UINT8 *)FixedPcdGetPtr (PcdDefault3PDb),
+    .DbSize            = (CONST UINT32)FixedPcdGetSize (PcdDefault3PDb),
+    .DbxPtr            = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultDbx),
+    .DbxSize           = (CONST UINT32)FixedPcdGetSize (PcdDefaultDbx),
+    .PkPtr             = (CONST UINT8 *)FixedPcdGetPtr (PcdDefaultPk),
+    .PkSize            = (CONST UINT32)FixedPcdGetSize (PcdDefaultPk),
+    .DbtPtr            = NULL,
+    .DbtSize           = 0,
+  }
+};
+
+/**
+  Interface to fetch platform Secure Boot Certificates, each payload
+  corresponds to a designated set of db, dbx, dbt, KEK, PK.
+
+  @param[in]  Keys        Pointer to hold the returned sets of keys. The
+                          returned buffer will be treated as CONST and
+                          permanent pointer. The consumer will NOT free
+                          the buffer after use.
+  @param[in]  KeyCount    The number of sets available in the returned Keys.
+
+  @retval     EFI_SUCCESS             The Keys are properly fetched.
+  @retval     EFI_INVALID_PARAMETER   Inputs have NULL pointers.
+  @retval     Others                  Something went wrong. Investigate further.
+**/
+EFI_STATUS
+EFIAPI
+GetPlatformKeyStore (
+  OUT SECURE_BOOT_PAYLOAD_INFO  **Keys,
+  OUT UINT8                     *KeyCount

```

Comment:
Why is this a `UINT8` instead of a `UINTN`?

Reply:
It is UINT8 in `Common\MU\MsCorePkg\Include\Library\SecureBootKeyStoreLib.h`, so to match the definition it is UINT8 here. Would you like it changed?

Reply:
Yes, it should be `UINTN` for an external facing API like this.

Reply:
I guess hold off. That will make this a breaking change, it is used quite a bit at the moment, and a series of breaking changes just went in people are still adjusting to.

Reply:
Sounds good. I'll add a calender event to change this in 2 weeks.